### PR TITLE
rpmbuild: fix mock config for building openio

### PR DESCRIPTION
**Pull request by jkasarherou**

### CODE REVIEW

File: openio-rpmbuild/build.py [hunk 1]
```diff
@@ -286,7 +286,7 @@ def patch_mock_config(distribution):
   with open(mock_cfg, 'rb') as fin:
     lines = fin.readlines()
     for line in lines:
-      if line.startswith('baseurl=http://mirror.openio.io'):
+      if upload_result and line.startswith('baseurl=http://mirror.openio.io'):
```
Comment (vincent-legoll) [inline openio-rpmbuild/build.py]: Is this trying to fix something ?
Please explain in the commit msg.
What's the rationale for putting this test in the loop ?

File: openio-rpmbuild/build.py [hunk 2]
```diff
@@ -300,9 +300,7 @@ def patch_mock_config(distribution):
 
 def mock(distribution, rpm_options, srpmsdir, upload_result):
   # FIXME: currently only doing this if uploading to oiorepo
-  if upload_result:
-      if urlparse.urlparse(upload_result).scheme == 'http':
```
Comment (vincent-legoll) [inline openio-rpmbuild/build.py]: Why did you remove this test ?

Comment (jkasarherou) [reply openio-rpmbuild/build.py]: in the case where you don't set an upload_result, the patch_mock_config is never called, which is bad, so I removed this test and I moved it (partially I admit) inside the patch_mock_config where it really belongs
